cpp/src/io/comp/io_uncomp.hpp

```diff
@@ -102,5 +102,19 @@ struct decompression_info {
 [[nodiscard]] size_t get_decompression_scratch_size(decompression_info const& di,
                                                     size_t num_buffers);
 
+/**
+ * @brief Computes the uncompressed sizes of Snappy-compressed input data.
+ *
+ * This function takes a collection of Snappy-compressed input data spans and computes
+ * their respective uncompressed sizes. The results are stored in the provided output span.
+ *
+ * @param inputs Compressed device memory buffers
+ * @param uncompressed_sizes Output device memory buffers to store the uncompressed sizes
+ * @param stream CUDA stream to be used for device operations and synchronization.
+ */
+void get_snappy_uncompressed_size(device_span<device_span<uint8_t const> const> inputs,
```

Comment:
It seems like some of the changes in this PR prefer the word "decompressed" instead of "uncompressed." This is using "uncompressed." Is there a technical difference or convention that I'm missing here?

Reply:
I'm trying to use "uncompressed" for data that is simply not compressed, and "decompressed" for data after, well, decompression. I do realize we're wildly inconsistent with this. For example, by this logic, this function should be named `get_snappy_decompressed_size`.
I'm open to suggestions as far as the impacted code in this PR is concerned. Ironing this in general will have to wait a second :)

Reply:
No worries. This can be dealt with later.